include/data_struct.h

```diff
@@ -197,6 +217,13 @@ class BaseLSTMStates {
     void reset_zeros();
 };
 
+// Smoother for linear layer when using LSTM
+class BaseSLinear {
```

Comment:
We don't need class for this case because it does not have the methods. Instead, we use `struct`